xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -1920,94 +1920,92 @@ void COutput::InitCycle()
     {
-      delete m_pp;
-      m_pp = NULL;
-      DropVppProcessedPictures();
-      m_config.processInfo->SetVideoDeintMethod("unknown");
+      CLog::Log(LOGINFO, LOGVIDEO, "VAAPI output: Initializing ffmpeg postproc");
+      m_pp = new CFFmpegPostproc();
     }
-    if (!m_pp)
+    else if ((method == VS_INTERLACEMETHOD_NONE && alwaysInsertVpp) || method == VS_INTERLACEMETHOD_VAAPI_BOB || method == VS_INTERLACEMETHOD_VAAPI_MACI || method == VS_INTERLACEMETHOD_VAAPI_MADI)
     {
-      if (method == VS_INTERLACEMETHOD_DEINTERLACE ||
-          method == VS_INTERLACEMETHOD_RENDER_BOB)
-      {
-        m_pp = new CFFmpegPostproc();
-        m_config.stats->SetVpp(false);
-      }
-      else
-      {
-        m_pp = new CVppPostproc();
-        m_config.stats->SetVpp(true);
-      }
-      if (m_pp->PreInit(m_config))
-      {
-        m_pp->Init(method);
-
-        if (method == VS_INTERLACEMETHOD_DEINTERLACE)
-          m_config.processInfo->SetVideoDeintMethod("yadif");
-        else if (method == VS_INTERLACEMETHOD_RENDER_BOB)
-          m_config.processInfo->SetVideoDeintMethod("render-bob");
-        else if (method == VS_INTERLACEMETHOD_VAAPI_BOB)
-          m_config.processInfo->SetVideoDeintMethod("vaapi-bob");
-        else if (method == VS_INTERLACEMETHOD_VAAPI_MADI)
-          m_config.processInfo->SetVideoDeintMethod("vaapi-madi");
-        else if (method == VS_INTERLACEMETHOD_VAAPI_MACI)
-          m_config.processInfo->SetVideoDeintMethod("vaapi-mcdi");
-      }
-      else
-      {
-        delete m_pp;
-        m_pp = NULL;
-      }
+      CLog::Log(LOGINFO, LOGVIDEO, "VAAPI output: Initializing vaapi postproc");
+      m_pp = new CVppPostproc();
+      m_config.stats->SetVpp(true);
     }
-  }
-  // progressive
-  else
-  {
-    method = VS_INTERLACEMETHOD_NONE;
-    if (m_pp && !m_pp->UpdateDeintMethod(method))
+    else
     {
-      delete m_pp;
-      m_pp = NULL;
-      DropVppProcessedPictures();
+      CLog::Log(LOGINFO, LOGVIDEO, "VAAPI output: Initializing skip postproc");
+      m_pp = new CSkipPostproc();
     }
-    if (!m_pp)
+
+    if (m_pp->PreInit(m_config))
+    {
+      m_pp->Init(method);
+    }
+    else
     {
+      CLog::Log(LOGERROR, "VAAPI output: Postproc preinit failed, using skip postproc as fallback");
+      delete m_pp;
+      m_pp = new CSkipPostproc();
+      method = VS_INTERLACEMETHOD_NONE;
       m_config.stats->SetVpp(false);
-      if (!CServiceBroker::GetSettings().GetBool(CSettings::SETTING_VIDEOPLAYER_PREFERVAAPIRENDER))
-        m_pp = new CFFmpegPostproc();
-      else
-      {
-        m_pp = new CSkipPostproc();
-        m_config.stats->SetVpp(true);
-      }
+
       if (m_pp->PreInit(m_config))
-      {
         m_pp->Init(method);
-        m_config.processInfo->SetVideoDeintMethod("none");
-      }
-      else
-      {
-        delete m_pp;
-        m_pp = NULL;
-      }
     }
   }
-  if (!m_pp) // fallback
+
+  switch (method)
```

Comment:
I don't like this change at all. I want to keep deinterlacing and progressive in separate paths like it was before.

Reply:
Then you have to tell me where I should differentiate. I assume c&p-ing the same code to a if and else branch will not make you happy :-)

Reply:
> I assume c&p-ing the same code to a if and else branch will not make you happy :-)

you might be wrong. I don't mind some duplicate lines of code if the design is clear.